presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentControlUtils.java

```diff
@@ -0,0 +1,220 @@
+ */
+package com.teradata.prestomanager.agent;
+
+import com.google.gson.Gson;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import javax.ws.rs.core.Response;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static com.teradata.prestomanager.agent.AgentFileUtils.getFileProperty;
+import static com.teradata.prestomanager.agent.CommandExecutor.executeCommand;
+import static com.teradata.prestomanager.common.RequestBuilder.newRequestBuilder;
+import static java.lang.String.format;
+import static javax.ws.rs.client.Entity.entity;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON_TYPE;
+import static javax.ws.rs.core.Response.Status.ACCEPTED;
+import static javax.ws.rs.core.Response.Status.INTERNAL_SERVER_ERROR;
+import static javax.ws.rs.core.Response.Status.NOT_FOUND;
+import static javax.ws.rs.core.Response.Status.OK;
+
+public final class AgentControlUtils
+{
+    private static final Logger LOGGER = LogManager.getLogger(AgentControlUtils.class);
+    private static final int WAIT_TIME_SECONDS = 60;
+    private static final int STEP_MILLISECONDS = 100;
+    private static final String LAUNCHER_SCRIPT = "/usr/lib/presto/bin/launcher";
+    private static final Path PATH_TO_CONFIG_FILE = Paths.get("/etc/presto/config.properties");
+
+    private AgentControlUtils() {}
+
+    public static Response startPresto()
+    {
+        if (!isInstalled()) {
+            return Response.status(NOT_FOUND).entity("Presto is not installed").build();
+        }
+        new Thread(() -> {
+            try {
+                int startPresto = executeCommand(120, "service", "presto", "startPresto");
+                if (startPresto != 0) {
+                    LOGGER.error("Failed to startPresto presto");
+                }
+            }
+            catch (PrestoManagerException e) {
+                LOGGER.error(e.getMessage(), e.getCause());
+            }
+            catch (Exception e) {
+                LOGGER.error("Unhandled exception", e);
+            }
+        }).start();
+        return Response.status(ACCEPTED).entity("Check status to make sure that presto has started").build();
+    }
+
+    public static Response stopPresto(StopType stopType)
+    {
+        if (!isInstalled()) {
+            return Response.status(NOT_FOUND).build();
+        }
+        if (!isRunning()) {
+            LOGGER.debug("Presto is not running");
+            return Response.status(OK).build();
+        }
+        try {
+            stopHelper(stopType);
+        }
+        catch (PrestoManagerException e) {
+            LOGGER.error(e.getMessage(), e.getCause());
+            return Response.status(INTERNAL_SERVER_ERROR).build();
+        }
+        catch (Exception e) {
+            LOGGER.error("Unhandled exception", e);
+            return Response.status(INTERNAL_SERVER_ERROR).build();
```

Comment:
Ditto, also may not be needing this as the method does not throw other type of exception